### PR TITLE
CODE-206 Rework of anchor day/date logic, primary shopping date, tests adapted

### PR DESCRIPTION
**Pull request by AkagiShigeru**

### CODE REVIEW

**Comment by AkagiShigeru** [comment]
There are some comments in the code that highlight the logic behind the anchor_day a bit more, please let me know if you agree with how it is implemented or if we can improve/change anything.

File: sous-chef/sous_chef/date/get_due_date.py [hunk 1]
```diff
@@ -31,12 +31,13 @@ class MealTime(ExtendedEnum):
 @dataclass
 class DueDatetimeFormatter:
     anchor_day: str
+    week_offset: int = 1
```
Comment (ArBridgeman) [inline sous-chef/sous_chef/date/get_due_date.py:34]: Why not use config value and no default value?

File: sous-chef/sous_chef/date/get_due_date.py [hunk 2]
```diff
@@ -77,14 +78,27 @@ def replace_time_with_meal_time(
...
+        # by default (week_offset=1) anchor_day will be in the upcoming week,
+        # independent of the current weekday.
+        # week_offset can be set to an arbitrarily higher (or smaller) number.
+        # an anchor day in the same week can be enforced with week_offset=0
+        # unless that day has already passed,
+        # then it will switch to the next week
+        # = same behaviour as with week_offset=1
+        # if a past anchor day is really desired (not an expected use case),
+        # a negative value of week_offset can be specified
+        if (today_index > weekday_index) and week_offset < 1:
```
Comment (ArBridgeman) [inline sous-chef/sous_chef/date/get_due_date.py:97]: Why not using self.week_offset here?

**Review by ArBridgeman — CHANGES_REQUESTED** [review]

File: sous-chef/sous_chef/date/get_due_date.py (hunk 1 above)
Comment (AkagiShigeru) [reply sous-chef/sous_chef/date/get_due_date.py:34]: We could remove the default and always specify it if you prefer that. I wouldn't see any problems with that. My slight preference in such situations is to have a default value that reflects the desired standard behavior.

File: sous-chef/sous_chef/date/get_due_date.py (hunk 2 above)
Comment (AkagiShigeru) [reply sous-chef/sous_chef/date/get_due_date.py:97]: This is in a static method.

File: sous-chef/sous_chef/date/get_due_date.py (hunk 1 above)
Comment (ArBridgeman) [reply sous-chef/sous_chef/date/get_due_date.py:34]: put a general config & then get fill these values in the post_init. then =1 not needed as needs to come from yaml...required for most things

File: sous-chef/sous_chef/date/get_due_date.py (hunk 2 above)
Comment (ArBridgeman) [reply sous-chef/sous_chef/date/get_due_date.py:97]: change to non-static method please & fix test if needed

File: sous-chef/tests/unit_tests/date/test_get_due_date.py [hunk 3]
```diff
@@ -97,7 +98,9 @@ def test_get_anchor_date(
         anchor_day, week_offset, expected_day, expected_index
     ):
         anchor_date = DueDatetimeFormatter(
```
Comment (ArBridgeman) [inline sous-chef/tests/unit_tests/date/test_get_due_date.py:100]: why not just read the file like we do for other services?

**Review by ArBridgeman — APPROVED** [review]

**Comment by ArBridgeman** [comment]
look at todoist comment & see if related potential bug has been resolved or not.
if not, please try to resolve it.

**Comment by AkagiShigeru** [comment]
> @AkagiShigeru look at todoist comment & see if related potential bug has been resolved or not. if not, please try to resolve it.

That should be resolved with the latest update.

**Review by ArBridgeman — APPROVED** [review]

**Pull request merged**